lib/compat/wordpress-6.3/template-parts-preloading.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Patches resources loaded by the block editor page
+ * to include Template Part posts.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Preloads requests needed for common Template Part posts
+ *
+ * @param array $preload_paths    Preload paths to be filtered.
+ * @param array $context          Context for the preload paths.
+ * @return array    the preload paths.
+ */
+function gutenberg_preload_template_parts( $preload_paths, $context ) {
+
+	// Limit to the Site Editor.
+	if ( ! empty( $context->name ) && 'core/edit-site' !== $context->name ) {
+		return $preload_paths;
+	}
+
+	// Get any template parts defined in theme.json.
+	$theme_json_template_parts = WP_Theme_JSON_Resolver::get_merged_data()->get_template_parts();
```

Comment:
@anton-vlasenko @ironprogrammer Do you have any idea of how I might be able to mock `WP_Theme_JSON_Resolver` in a test? I want to write tests but not sure how I could make that testable.